### ISSUE TITLE
refactor to universal library?

### ISSUE THREAD

**Issue by Pomax** [poilu/raphael-boolean#7]
I know it's been over two years since you worked on this, but I was wondering if you might be interested in converting this code a universal library that just operates on "SVG strings" rather than Raphael structures, since there are no boolean shape libraries out there, and it would be incredibly valuable for this code to work with "any SVG path string" rather than with Raphael-object wrapped strings.

**Comment by poilu** [comment]
That's an interesting idea. Before I started this plugin I was wondering myself, that there is no javascript library providing this features. It's astonishing, that there still isn't one, although SVG is such a mature standard.
I guess if you say it should  'work with "any SVG path string"' it should be completely independent from raphael library. Developing in this direction has a big challenge: the plugin relies on raphael utility methods that provide some fundamental functionality. This is mainly finding dots on curves, finding path intersections, getting the bounding box of a path and check if a point is inside a path. As you probably know, this is far from being trivial and we would need to find a way to extract this functionality from raphael core or find another library, which provide such methods. 
Beside that, in some special cases raphael.isPointInsidePath produced errors, which I never were able to get rid of (see issue https://github.com/poilu/raphael-boolean/issues/3). So there are some additional problems to solve.

**Comment by Pomax** [comment]
fair enough. I'm equally amazed (and often angry =) over the fact there still isn't a general purpose boolean shape library in JS... I've needed one so often, and there just aren't any. So far, this one is still the most functional I've been able to find.
